cmd/kubeadm/app/apis/kubeadm/types.go

```diff
@@ -263,9 +262,10 @@ type LocalEtcd struct {
 
 	// ExtraArgs are extra arguments provided to the etcd binary
 	// when run inside a static pod.
-	// A key in this map is the flag name as it appears on the
-	// command line except without leading dash(es).
-	ExtraArgs map[string]string
+	// An argument name in this list is the flag name as it appears on the
+	// command line except without leading dash(es). Extra arguments will override existing
+	// default arguments. Duplicate extra arguments are allowed.
+	ExtraArgs []Arg
```

Comment:
Do we have any testcase to test that config file now allows the arg with duplicated name? I haven't seen it.

Reply:
i don't think we need to test it on the config structs directly, but we do test TestArgumentsToCommand, which is the important function that produces a command with --flags.